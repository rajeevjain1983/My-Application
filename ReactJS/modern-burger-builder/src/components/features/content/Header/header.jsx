import React from "./node_modules/react";
import withStyle from "../../../common/hoc/withStyle";
import styles from "./header.style";

const header = (props) => {
  console.log("header");
  return (
    <div className={props.className}>
      <button className="buttonStyle" onClick={props.toggleTheme}>
        Toggle Theme
      </button>
    </div>
  );
};

export default withStyle(header, styles);
export { header as headerVanilla };
