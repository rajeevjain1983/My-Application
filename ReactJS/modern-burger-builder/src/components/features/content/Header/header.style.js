import { css } from "./node_modules/styled-components";

const styles = css`
  background-image: linear-gradient(
    to right,
    ${(props) => props.theme.primaryColor},
    ${(props) => props.theme.secondaryColor}
  );
  color: ${(props) => props.theme.secondaryColor};
  padding: 10px;
  margin: 50px;
  justify-content: flex-end;
  display: flex;
  .buttonStyle {
    border: 1px solid ${(props) => props.theme.primaryColor};
    border-radius: 30px;
    padding: 5px 10px;
  }
`;

export default styles;
