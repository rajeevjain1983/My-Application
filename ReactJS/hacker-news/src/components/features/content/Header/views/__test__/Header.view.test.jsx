import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Header from "../Header.view";

configure({ adapter: new Adapter() });

describe("<Header>", () => {
  it("Render Header view", () => {
    const component = shallow(<Header />);

    expect(component).toMatchSnapshot();
  });
});
