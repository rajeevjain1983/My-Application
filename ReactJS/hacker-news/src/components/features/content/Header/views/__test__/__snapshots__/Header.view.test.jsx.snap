// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> Render Header view 1`] = `
<Header
  className="sc-AxjAm dJrOAM"
/>
`;
